### ISSUE TITLE
updateTransform error, due to removeChild changes this.children.length

### ISSUE THREAD

**Issue by kambing86** [GoodBoyDigital/pixi.js#923]
Scenario:

When one movieClip is not loop, after gotoAndPlay method is called and hit the last frame
movieClip.onComplete is triggered;

```javascript
movieClip.onComplete = function () {
    //movieClip.gotoAndStop(0); //adding this line of code couldn't help the problem
    if (movieClip.parent)
        container.removeChild(movieClip);
}
```

the container still have the possibility doing updateTransform inside the loop of wrong length, which makes no sense.

Error occured as below:

> Uncaught TypeError: Cannot read property 'updateTransform' of undefined pixi.dev.js:1498
> PIXI.DisplayObjectContainer.updateTransform pixi.dev.js:1498
> PIXI.DisplayObjectContainer.updateTransform pixi.dev.js:1498
> PIXI.DisplayObjectContainer.updateTransform pixi.dev.js:1498
> PIXI.Stage.updateTransform pixi.dev.js:4078
> PIXI.WebGLRenderer.render pixi.dev.js:6575

```javascript
PIXI.DisplayObjectContainer.prototype.updateTransform = function()
{
    //this._currentBounds = null;

    if(!this.visible)return;

    PIXI.DisplayObject.prototype.updateTransform.call( this );

    if(this._cacheAsBitmap)return;

    for(var i=0,j=this.children.length; i<j; i++)
    {
        this.children[i].updateTransform(); <- ERROR OCCURRED HERE
    }
};
```

should we just remove the variable `j`?
even though by looking into `this.children.length` every iteration will reduce the performance.

Any solution for this issue?

**Comment by GoodBoyDigital** [comment]
This has been addressed for v3. :+1:

**Comment by xPaw** [comment]
This doesn't appear to be fixed in 3.0.6 as I just ran into this issue, had to use setTimeout with delay of 0.

**Comment by suroorwijdan** [comment]
Having the same issue here, using version 3.0.9
